### PR TITLE
Add and fix size restrictions to targets

### PR DESCRIPTION
**Pull request by esajaa**
### Summary of changes

Added size restrictions to several boards, and also fixed some previous ones that were too large or too small.

Used target.restrict_size calculation method:
For 1MB bank: Hex 0x100000 (=Dec 1024 * Dec 1024) - target.app_offset = target.restrict_size
For 512KB bank: Hex 0x080000 (=Dec 512 * Dec 1024) - target.app_offset = target.restrict_size

[x] I confirm this contribution is my own and I agree to license it with Apache 2.0.

[x] I confirm the moderators may change the PR before merging it in.

For new board enablements only:

[] I confirm the board is [Mbed Enabled](https://www.mbed.com/en/about-mbed/mbed-enabled/introduction/) and passes the Mbed Enabled test set.

[] I confirm the contribution has been tested properly and the tests results for [TESTS](https://github.com/ARMmbed/mbed-os-example-pelion/tree/master/TESTS) are attached.

[] I confirm `mbed-os.lib` and `mbed-cloud-client.lib` hashes or the content in folders `mbed-os` and `mbed-cloud-client` were not modified in order to pass the tests.

- Maintainers of this repository update the Mbed OS and Client hashes.
- Please report any issues through issues in relevant repositories.

### CODE REVIEW

**Comment by iot-pdmc** [comment]
Thanks for your changes.
@ARMmbed/pdmc-integrators please review.

**Comment by teetak01** [comment]
run-ci

**Comment by iot-pdmc** [comment]
# [Test run: SUCCESS](https://jenkins-client.isgtesting.com/job/mbed-os-example-pelion/611/)
***

**Test summary:**
Executed: 24
Failed: 0
Skipped: 4
Errors: 0
[Build artifacts](http://pdmc-ci.s3-website-eu-west-1.amazonaws.com/?prefix=jenkins-ci/mbed-os-example-pelion/611)

posted-by-tag: https://jenkins-client.isgtesting.com/job/mbed-os-example-pelion/

**Comment by MarceloSalazar** [comment]
Thanks - looks good
I've added @maclobdell @toyowata to be aware of this.
I assume the compile test would be sufficient to verify that there are no regressions.

**Comment by teetak01** [comment]
yes, this is purely compile-time check to verify that the application does not overflow to bank2.

File: mbed_app.json [hunk 1]
```diff
@@ -467,6 +470,7 @@
             "target.bootloader_img"                     : "bootloader/mbed-bootloader-gr_peach-internal_flash-no_rot-v4.1.2.bin",
             "target.header_offset"                      : "0x10000",
             "target.app_offset"                         : "0x10400",
+            "target.restrict_size"                      : "0xEFC00",
```
Comment (toyowata) [inline mbed_app.json]: Renesas [GR-PEACH](https://os.mbed.com/platforms/Renesas-GR-PEACH/) (RZ_A1H) has 8MB flash memory, so 4MB per bank (not 1MB).
https://github.com/ARMmbed/mbed-os-example-pelion/blob/32354d030e7e4bb87bd2f83c9f83adc60ab23065/mbed_app.json#L477

`target.restrict_size` should be `0x3EFC00`

File: mbed_app.json [hunk 2]
```diff
@@ -486,6 +490,7 @@
             "target.bootloader_img"                     : "bootloader/mbed-bootloader-gr_lychee-internal_flash-no_rot-v4.1.2.bin",
             "target.header_offset"                      : "0x10000",
             "target.app_offset"                         : "0x10400",
+            "target.restrict_size"                      : "0xEFC00",
```
Comment (toyowata) [inline mbed_app.json]: Same as GR-PEACH.  It should be `0x3EFC00`

File: mbed_app.json (hunk 1 above)
Comment (esajaa) [reply mbed_app.json]: Good catch, fixed

**Comment by teetak01** [comment]
@esajaa rebase & merge.

**Comment by teetak01** [comment]
build-only

**Comment by iot-pdmc** [comment]
# [Test run: SUCCESS](https://jenkins-client.isgtesting.com/job/mbed-os-example-pelion/612/)
***
[Build artifacts](http://pdmc-ci.s3-website-eu-west-1.amazonaws.com/?prefix=jenkins-ci/mbed-os-example-pelion/612)

posted-by-tag: https://jenkins-client.isgtesting.com/job/mbed-os-example-pelion/

**Pull request merged**